crates/sncast/src/starknet_commands/account/create.rs

```diff
@@ -127,15 +142,30 @@ async fn generate_account(
     provider: &JsonRpcClient<HttpTransport>,
     salt: FieldElement,
     class_hash: FieldElement,
+    account_type: &AccountType,
 ) -> Result<(serde_json::Value, FieldElement)> {
+    let chain_id = get_chain_id(provider).await?;
     let private_key = SigningKey::from_random();
+    let signer = LocalWallet::from_signing_key(private_key.clone());
 
-    let address: FieldElement = get_contract_address(
-        salt,
-        class_hash,
-        &[private_key.verifying_key().scalar()],
-        FieldElement::ZERO,
-    );
+    let (address, fee_estimate) = match account_type {
+        AccountType::Oz => {
+            let factory =
+                OpenZeppelinAccountFactory::new(class_hash, chain_id, signer, provider).await?;
+            get_address_and_deployment_fee(factory, salt).await?
+        }
+        AccountType::Argent => {
+            let factory = ArgentAccountFactory::new(
+                class_hash,
+                chain_id,
+                FieldElement::ZERO,
+                signer,
+                provider,
+            )
+            .await?;
+            get_address_and_deployment_fee(factory, salt).await?
+        }
```

Comment:
what happens if someone tries to create braavos atm? 
I suppose we should add a new case in here that will gather all the other possible AccountTypes and return some error

Reply:
probably applicable to other, similar match statements

Reply:
> what happens if someone tries to create braavos atm?

It is not possible to pass `--type braavos` because it's an enum, and the only permitted values are `oz` and `argent` atm. If someone attempts to pass a braavos class hash with one of these allowable types, it would result in a failure during deployment.

Reply:
ok, so the clap handles it then, that's great :+1: